linux/base.Dockerfile

```diff
@@ -21,6 +21,8 @@ COPY linux/tdnfinstall.sh .
 RUN tdnf update -y --refresh && \
   bash ./tdnfinstall.sh \
   azurelinux-repos-extended && \
+  curl -sSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/pki/rpm-gpg/microsoft.gpg && \
```

Comment:
This can be done similar to `azurelinux-repos-extended` line 23 via package installation: azurelinux-repos-ms-non-oss-3.0 

https://packages.microsoft.com/azurelinux/3.0/prod/base/x86_64/Packages/a/